src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,93 +1,213 @@
-            if (!span.IntersectsWith(textSpan))
+            return true;
+        }
+
+        protected override SyntaxList<SyntaxNode> GetStatements(SyntaxNode node)
+        {
+            switch (node)
             {
-                return null;
+                case BlockSyntax n:
+                    return n.Statements;
+                case SwitchSectionSyntax n:
+                    return n.Statements;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(node);
             }
-
-            return ifStatement;
         }
 
-        protected override SyntaxNode GetRootWithInvertIfStatement(
-            Document document,
-            SemanticModel model,
-            SyntaxNode ifStatementSyntax,
-            CancellationToken cancellationToken)
+        protected override int GetJumpStatementRawKind(SyntaxNode node)
         {
-            var generator = SyntaxGenerator.GetGenerator(document);
-            var syntaxFacts = GetSyntaxFactsService();
+            switch (node.Kind())
+            {
+                case SyntaxKind.SwitchSection:
+                    return (int)SyntaxKind.BreakStatement;
+
+                case SyntaxKind.LocalFunctionStatement:
+                case SyntaxKind.SetAccessorDeclaration:
+                case SyntaxKind.GetAccessorDeclaration:
+                case SyntaxKind.AddAccessorDeclaration:
+                case SyntaxKind.RemoveAccessorDeclaration:
+                case SyntaxKind.MethodDeclaration:
+                case SyntaxKind.ConstructorDeclaration:
+                case SyntaxKind.DestructorDeclaration:
+                case SyntaxKind.OperatorDeclaration:
+                case SyntaxKind.ConversionOperatorDeclaration:
+                case SyntaxKind.AnonymousMethodExpression:
+                case SyntaxKind.SimpleLambdaExpression:
+                case SyntaxKind.ParenthesizedLambdaExpression:
+                    return (int)SyntaxKind.ReturnStatement;
 
-            var oldIfStatement = (IfStatementSyntax)ifStatementSyntax;
+                case SyntaxKind.DoStatement:
+                case SyntaxKind.WhileStatement:
+                case SyntaxKind.ForStatement:
+                case SyntaxKind.ForEachStatement:
+                case SyntaxKind.ForEachVariableStatement:
+                    return (int)SyntaxKind.ContinueStatement;
+            }
 
-            // For single line statement, we swap the TrailingTrivia to preserve the single line
-            StatementSyntax newIfNodeStatement = null;
-            ElseClauseSyntax newElseStatement = null;
+            return -1;
+        }
 
-            var hasNewLineAfterClosingBrace = oldIfStatement.Statement.GetTrailingTrivia().Any(trivia => trivia.Kind() == SyntaxKind.EndOfLineTrivia);
-            if (hasNewLineAfterClosingBrace)
+        protected override SyntaxNode GetJumpStatement(int rawKind)
+        {
+            switch ((SyntaxKind)rawKind)
```

Comment:
curious, how is trivia handled?

Reply:
node seems not C# specific? does vb have vb specific ones? otherwise, put these in syntax generator and have this in shared base type?

Reply:
>curious, how is trivia handled?

It's not. I'm looking for ways to handle this. any insight you could provide to make it as generic as possible is welcome.

Reply:
>node seems not C# specific? does vb have vb specific ones? otherwise, put these in syntax generator and have this in shared base type?

Yes, we create the a syntax node given the syntaxkind. I wonder if there is a helper for that already? (probably not, because this method is pretty narrow in the input and output).

Reply:
we would like to create generic tiriva handling system as well. but we dont have one right now.

Reply:
Since we don't have anything right now, let's just take a good-enough approach.  If customers find this experience unsatisfactory, we can address specific bugs that come up.